src/runner/driver-test-runner.js

```diff
@@ -288,10 +288,8 @@ export class DriverTestRunner {
   }
 
   _appendBaseUrl(pathname) {
-    // protocol ends with a ':' and pathname starts with a '/'
-    const base = `${this.baseUrl.protocol}://${this.baseUrl.hostname}:${this.baseUrl.port}${this.baseUrl.pathname}`;
     const newPath = `${this.baseUrl.pathname ? `${this.baseUrl.pathname}/` : ''}${pathname}`;
-    return new URL(newPath, base);
+    return new URL(newPath, this.baseUrl.toString());
```

Comment:
@gnarf I was able to simplify this method a bit, since we don't really need to replicate the `URL.toString()` method in #L291-292.